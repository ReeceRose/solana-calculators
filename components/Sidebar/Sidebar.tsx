import { useState } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';

export default function Sidebar(): JSX.Element {
  const [collapseShow, setCollapseShow] = useState('hidden');
  const router = useRouter();

  return (
    <>
      <nav className="relative z-10 flex flex-wrap items-center justify-between px-6 py-4 bg-gray-100 shadow-xl md:left-0 md:block md:fixed md:top-0 md:bottom-0 md:overflow-y-auto md:flex-row md:flex-nowrap md:overflow-hidden md:w-64">
        <div className="flex flex-wrap items-center justify-between w-full px-0 mx-auto md:flex-col md:items-stretch md:min-h-full md:flex-nowrap">
          <button
            className="px-3 py-1 text-xl leading-none text-black bg-transparent border border-transparent border-solid rounded opacity-50 cursor-pointer md:hidden"
            type="button"
            onClick={() => setCollapseShow('bg-white m-2 py-3 px-6')}
          >
            <i className="fas fa-bars"></i>
          </button>
          <Link href="/">
            <a
              href="#"
              className="inline-block p-4 px-0 mr-0 text-sm font-bold text-left text-gray-600 uppercase md:block md:pb-2 whitespace-nowrap"
            >
              Solana Calculators
            </a>
          </Link>

          <div
            className={
              'md:flex md:flex-col md:items-stretch md:opacity-100 md:relative md:mt-4 md:shadow-none shadow absolute top-0 left-0 right-0 z-40 overflow-y-auto overflow-x-hidden h-auto items-center flex-1 rounded ' +
              collapseShow
            }
          >
            <div className="block pb-4 mb-4 border-b border-gray-200 border-solid md:min-w-full md:hidden">
              <div className="flex flex-wrap">
                <div className="w-6/12">
                  <Link href="/">
                    <a
                      href="#pablo"
                      className="inline-block p-4 px-0 mr-0 text-sm font-bold text-left text-gray-600 uppercase md:block md:pb-2 whitespace-nowrap"
                    >
                      Solana Calculators
                    </a>
                  </Link>
                </div>
                <div className="flex justify-end w-6/12">
                  <button
                    type="button"
                    className="px-3 py-1 text-xl leading-none text-black bg-transparent border border-transparent border-solid rounded opacity-50 cursor-pointer md:hidden"
                    onClick={() => setCollapseShow('hidden')}
                  >
                    <i className="fas fa-times"></i>
                  </button>
                </div>
              </div>
            </div>

            <hr className="my-4 md:min-w-full" />
            <h6 className="block pt-1 pb-4 text-xs font-bold text-gray-600 no-underline uppercase md:min-w-full">
              Calculators
            </h6>

            <ul className="flex flex-col list-none md:flex-col md:min-w-full">
              <li className="items-center">
                <Link href="/calculators/rent-calculator">
                  <a
                    className={
                      'text-xs uppercase py-3 font-bold block ' +
                      (router.pathname.indexOf(
                        '/calculators/rent-calculator'
                      ) !== -1
                        ? 'text-blue-500 hover:text-blue-600'
                        : 'text-gray-600 hover:text-gray-500')
                    }
                  >
                    <i
                      className={
                        'fas fa-calculator mr-2 text-sm ' +
                        (router.pathname.indexOf(
                          '/calculators/rent-calculator'
                        ) !== -1
                          ? 'opacity-75'
                          : 'text-gray-600')
                      }
                    ></i>{' '}
                    Rent Calculator
                  </a>
                </Link>
              </li>
            </ul>

            <hr className="my-4 md:min-w-full" />
            <h6 className="block pt-1 pb-4 text-xs font-bold text-gray-600 no-underline uppercase md:min-w-full">
              About
            </h6>

            <ul className="flex flex-col list-none md:flex-col md:min-w-full">
              <li className="items-center">
                <Link href="https://github.com/reecerose/solana-calculators">
                  <a
                    target="_blank"
                    rel="noreferrer"
                    className={
                      'text-xs uppercase py-3 font-bold block text-gray-600 hover:text-gray-500'
                    }
                  >
                    <i
                      className={'fab fa-github mr-2 text-sm text-gray-600'}
                    ></i>{' '}
                    GitHub
                  </a>
                </Link>
              </li>
            </ul>

            <hr className="my-4 md:min-w-full" />
            <h6 className="block pt-1 pb-4 text-xs font-bold text-gray-600 no-underline uppercase md:min-w-full">
              Copyright Â© {new Date().getFullYear()} <br />
              <Link href="https://github.com/ReeceRose/">
                <a target="_blank" rel="noreferrer">
                  Reece Rose
                </a>
              </Link>
            </h6>
          </div>
        </div>
      </nav>
    </>
  );
}
