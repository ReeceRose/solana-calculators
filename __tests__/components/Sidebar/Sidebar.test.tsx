/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';

import Sidebar from '../../../components/Sidebar/Sidebar';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
}));

test('Sidebar component matches snapshot', () => {
  const { asFragment } = render(<Sidebar />);
  expect(asFragment()).toMatchSnapshot();
});
