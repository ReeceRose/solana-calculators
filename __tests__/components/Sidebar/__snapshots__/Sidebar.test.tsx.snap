// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="relative z-10 flex flex-wrap items-center justify-between px-6 py-4 bg-gray-100 shadow-xl dark:bg-gray-900 md:left-0 md:block md:fixed md:top-0 md:bottom-0 md:overflow-y-auto md:flex-row md:flex-nowrap md:overflow-hidden md:w-64"
  >
    <div
      class="flex flex-wrap items-center justify-between w-full px-0 mx-auto md:flex-col md:items-stretch md:min-h-full md:flex-nowrap"
    >
      <button
        class="px-3 py-1 text-xl leading-none text-black bg-transparent border border-transparent border-solid rounded opacity-50 cursor-pointer md:hidden"
        type="button"
      >
        <i
          class="fas fa-bars"
        />
      </button>
      <a
        class="inline-block p-4 px-0 mr-0 text-sm font-bold text-left text-gray-600 uppercase hover:text-gray-500 hover:dark:text-gray-300 dark:text-gray-50 md:block md:pb-2 whitespace-nowrap"
        href="#"
      >
        Solana Calculators
      </a>
      <div
        class="md:flex md:flex-col md:items-stretch md:opacity-100 md:relative md:mt-4 md:shadow-none shadow absolute top-0 left-0 right-0 z-40 overflow-y-auto overflow-x-hidden h-auto items-center flex-1 rounded hidden"
      >
        <div
          class="block pb-4 mb-4 border-b border-gray-200 border-solid md:min-w-full md:hidden"
        >
          <div
            class="flex flex-wrap"
          >
            <div
              class="w-6/12"
            >
              <a
                class="inline-block p-4 px-0 mr-0 text-sm font-bold text-left text-gray-600 uppercase hover:dark:text-gray-300 hover:text-gray-500 dark:text-gray-50 md:block md:pb-2 whitespace-nowrap"
                href="#pablo"
              >
                Solana Calculators
              </a>
            </div>
            <div
              class="flex justify-end w-6/12"
            >
              <button
                class="px-3 py-1 text-xl leading-none text-black bg-transparent border border-transparent border-solid rounded opacity-50 cursor-pointer md:hidden"
                type="button"
              >
                <i
                  class="fas fa-times"
                />
              </button>
            </div>
          </div>
        </div>
        <hr
          class="my-4 md:min-w-full"
        />
        <h6
          class="block pt-1 pb-4 text-xs font-bold text-gray-600 no-underline uppercase dark:text-gray-50 md:min-w-full"
        >
          Calculators
        </h6>
        <ul
          class="flex flex-col list-none md:flex-col md:min-w-full"
        >
          <li
            class="items-center"
          >
            <a
              class="text-xs uppercase py-3 font-bold block text-blue-500 hover:text-blue-600"
              href="/calculators/rent-calculator"
            >
              <i
                class="fas fa-calculator mr-2 text-sm opacity-75"
              />
               Rent Calculator
            </a>
          </li>
        </ul>
        <hr
          class="my-4 md:min-w-full"
        />
        <h6
          class="block pt-1 pb-4 text-xs font-bold text-gray-600 no-underline uppercase dark:text-gray-50 md:min-w-full"
        >
          About
        </h6>
        <ul
          class="flex flex-col list-none md:flex-col md:min-w-full"
        >
          <li
            class="items-center"
          >
            <a
              class="text-xs uppercase py-3 font-bold block text-gray-600 hover:text-gray-500 dark:text-gray-50 hover:dark:text-gray-300"
              href="https://github.com/reecerose/solana-calculators"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-github mr-2 text-sm text-gray-600 dark:text-gray-50"
              />
               GitHub
            </a>
          </li>
        </ul>
        <hr
          class="my-4 md:min-w-full"
        />
        <h6
          class="block pt-1 pb-4 text-xs font-bold text-gray-600 no-underline uppercase dark:text-gray-50 md:min-w-full"
        >
          Copyright Â© 2021 
          <br />
          <a
            href="https://github.com/ReeceRose/"
            rel="noreferrer"
            target="_blank"
          >
            Reece Rose
          </a>
        </h6>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
